example/api.js

```diff
@@ -0,0 +1,233 @@
+      return result;
+    } else if (decoded instanceof Map) {
+      // Handle Map as previously explained
+      if (!decoded.has(-2) || !decoded.has(-3)) {
+        throw new Error('Unexpected CBOR structure');
+      }
+
+      const x = decoded.get(-2);
+      const y = decoded.get(-3);
+
+      const tag = (y[y.length - 1] & 1) === 1 ? 3 : 2;
+      const result = new Uint8Array(33);
+      result[0] = tag;
+      result.set(x, 1);
+
+      return result;
+    } else {
+      throw new Error('Unexpected decoded CBOR structure');
+    }
+  } catch (error) {
+    console.error('Error decoding CBOR data:', error);
+    throw new Error('Failed to decode CBOR data properly');
+  }
+};
+
+// Call this function to create passkeys
+export async function createPassKeys() {
+  const keyPair = await crypto.subtle.generateKey(
+    {
+      name: 'ECDSA',
+      namedCurve: 'P-256',
+    },
+    true,
+    ['sign', 'verify'],
+  );
+
+  const publicKey = await crypto.subtle.exportKey('raw', keyPair.publicKey);
+  const privateKey = await crypto.subtle.exportKey('pkcs8', keyPair.privateKey);
+
+  const encodedPublicKey = cbor.encode(new Uint8Array(publicKey));
+  return {
+    passKeyPrivateKey: u8aToHex(new Uint8Array(privateKey)),
+    passKeyPublicKey: u8aToHex(encodedPublicKey),
+  };
+}
+
+// Call this function to sign passkey challenges
+export async function signPasskeyChallenge(privateKeyHex, passkeyChallengeHex) {
+  const privateKey = await importPrivateKey(privateKeyHex);
+  const authenticatorDataRaw = 'WJ8JTNbivTWn-433ubs148A7EgWowi4SAcYBjLWfo1EdAAAAAA';
+  const replacedClientDataRaw =
+    'eyJ0eXBlIjoid2ViYXV0aG4uZ2V0IiwiY2hhbGxlbmdlIjoiI3JwbGMjIiwib3JpZ2luIjoiaHR0cHM6Ly9wYXNza2V5LmFtcGxpY2EuaW86ODA4MCIsImNyb3NzT3JpZ2luIjpmYWxzZSwiYWxnIjoiSFMyNTYifQ';
+  const challengeReplacer = '#rplc#';
+  let clientData = base64UrlToUint8Array(replacedClientDataRaw);
+  let authenticatorData = base64UrlToUint8Array(authenticatorDataRaw);
+  const passkeyChallengeBytes = hexToU8a(passkeyChallengeHex);
+
+  // Convert passkeyChallengeBytes to a base64 URL-encoded string
+  const calculatedChallengeBase64url = btoa(String.fromCharCode(...passkeyChallengeBytes))
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=+$/, '');
+
+  // Replace the placeholder in the clientData JSON string
+  const clientDataJSON = new TextDecoder().decode(clientData).replace(challengeReplacer, calculatedChallengeBase64url);
+  const clientDataBytes = new TextEncoder().encode(clientDataJSON);
+
+  // Prepare the signing payload
+  const shaClientData = await sha256(clientDataBytes);
+
+  // Concatenate authenticatorData and shaClientData
+  const dataToSign = new Uint8Array([...authenticatorData, ...shaClientData]);
+
+  // Sign the challenge using the private key
+  const passKeySignature = await crypto.subtle.sign(
+    {
+      name: 'ECDSA',
+      hash: { name: 'SHA-256' }, // Specify the hash algorithm used
+    },
+    privateKey,
+    dataToSign,
+  );
```

Comment:
I will confirm here, if i am double hashing here, because of this `hash: { name: 'SHA-256' }, // Specify the hash algorithm used`